com.unity.ml-agents/Runtime/SideChannels/EngineConfigurationChannel.cs

```diff
@@ -20,27 +20,21 @@ public EngineConfigurationChannel()
         }
 
         /// <inheritdoc/>
-        public override void OnMessageReceived(byte[] data)
+        public override void OnMessageReceived(IncomingMessage msg)
         {
-            using (var memStream = new MemoryStream(data))
-            {
-                using (var binaryReader = new BinaryReader(memStream))
-                {
-                    var width = binaryReader.ReadInt32();
-                    var height = binaryReader.ReadInt32();
-                    var qualityLevel = binaryReader.ReadInt32();
-                    var timeScale = binaryReader.ReadSingle();
-                    var targetFrameRate = binaryReader.ReadInt32();
+            var width = msg.ReadInt32();
+            var height = msg.ReadInt32();
+            var qualityLevel = msg.ReadInt32();
+            var timeScale = msg.ReadFloat32();
+            var targetFrameRate = msg.ReadInt32();
 
-                    timeScale = Mathf.Clamp(timeScale, 1, 100);
+            timeScale = Mathf.Clamp(timeScale, 1, 100);
```

Comment:
This is maybe for another PR, but the 100x restriction on timescale was removed in Unity somewhere in the 2018.X stream. I think we can safely get rid of it ourselves as well.

Reply:
Logged as https://jira.unity3d.com/browse/MLA-748 for followup